### PR TITLE
update the nanopore and fasta input channel- branch "bouncy-basenji"

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/taxprofiler.nf [hunk 1]
```diff
@@ -113,65 +113,63 @@ workflow TAXPROFILER {
...
-        if (meta.single_end && fastq_2) {
-            error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+            if ( !meta.is_fasta && !fastq_1 ) {
+                error("ERROR: Please check input samplesheet: entry `fastq_1` doesn't exist!")
+            }
+            if ( meta.instrument_platform == 'OXFORD_NANOPORE' && fastq_2 ) {
+                error("Error: Please check input samplesheet: for Oxford Nanopore reads entry `fastq_2` should be empty!")
+            }
+            if ( meta.single_end && fastq_2 ) {
+                error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
```
Comment (jfy133) [inline workflows/taxprofiler.nf:132]: I would rather not do validation within the `branch` itself, better to have a separate map function for doing this.

File: workflows/taxprofiler.nf [hunk 2]
```diff
@@ -113,65 +113,63 @@ workflow TAXPROFILER {
...
-        //
-        nanopore: instrument_platform == 'OXFORD_NANOPORE' && meta.single_end
-            return [meta, [fastq_1]]
-        fastq_pe: fastq_2
-            return [meta, [fastq_1, fastq_2]]
-        ch_fasta: meta.is_fasta && meta.single_end
-            return [meta, [fasta]]
-    }
-    .set { ch_input }
+        .set { ch_input }
```
Comment (jfy133) [inline workflows/taxprofiler.nf:145]: We agreed a long time ago within nf-core to not use `.set` but use the `ch_input = ` syntax at the beginning of the chain

File: workflows/taxprofiler.nf (hunk 1 above)
Comment (LilyAnderssonLee) [reply workflows/taxprofiler.nf:132]: I will re-organise this section.

File: workflows/taxprofiler.nf [hunk 3]
```diff
@@ -111,67 +111,67 @@ workflow TAXPROFILER {
...
+            }
+            if ( meta.single_end && fastq_2 ) {
+                error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+            }
+            return [ meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ]
         }
-        if (meta.single_end && fastq_2) {
-            error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+        .branch { meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ->
+            fastq_se: meta.single_end
```
Comment (jfy133) [inline workflows/taxprofiler.nf:137]: Any reason why to split fastq_pe and fastq_se when you immediately merge after?

File: workflows/taxprofiler.nf [hunk 4]
```diff
@@ -111,67 +111,67 @@ workflow TAXPROFILER {
...
+                return [ meta, [ fastq_1 ] ]
+            nanopore: instrument_platform == 'OXFORD_NANOPORE'
+                meta.single_end = true
+                return [ meta, [ fastq_1 ] ]
+            fastq_pe: fastq_2
+                return [ meta, [ fastq_1, fastq_2 ] ]
+            ch_fasta: meta.is_fasta
```
Comment (jfy133) [inline workflows/taxprofiler.nf:144]:
```suggestion
            fasta: meta.is_fasta
```

**Review by jfy133 — COMMENTED** [review]
I think a lot of these should be moved into the new pipeline validation subworkflow that weas added in the latest template, but this will do for now.

Last reminaing comments

File: workflows/taxprofiler.nf (hunk 3 above)
Comment (LilyAnderssonLee) [reply workflows/taxprofiler.nf:137]: Good question. I think I just followed the formal input_check logic without thinking. You are right, I will modify this part.

**Pull request merged**